lib/tesla/middleware/retry.ex

```diff
@@ -90,7 +92,10 @@ defmodule Tesla.Middleware.Retry do
     if context.should_retry.(res) do
       backoff(context.max_delay, context.delay, context.retries, context.jitter_factor)
       context = update_in(context, [:retries], &(&1 + 1))
-      retry(env, next, context)
+
+      res
+      |> context.update_env.(env)
```

Comment:
```suggestion
      |> context.before_retry.(env)
```

Hey, I am sorry it took me this long to help you.

Do you think it would be better to rename the callback to `before_retry`? Some people may not necessarily `update` the `env` but just log it or do something else.

What do you think?

Reply:
Not sure about this. It should still log the request itself, right? 🤔. My only concern here is that `before_retry` does not sound like a side effect where it modifies the env.

Anyway, I don't have a strong opinion here. I can change it to `before_retry` if it fits better.

Reply:
The intent of modifying or not is based on the API contracts. In other words, I give you an `env,` and I expect an `env` in return.

If that is a side-effect or not doesn't matter. You could have a rotating auth token, and your function may do a simple circular linked-list look-up that isn't a side effect per se.

Let's change it, I am unsure if I am right, but the function isn't around a particular action you expect the users to take. Just a simple callback that you get to do things with the `env` before retrying.

Time will tell 😄

Reply:
@yordis fixed.